ion-hash/src/lib.rs

```diff
@@ -57,62 +55,53 @@ impl Markers {
 /// This is the reason for this is we can't go from [`Digest`] to these
 /// traits (e.g. [`Update`]) but because of a blanket implementation, we can go
 /// the other way.
-pub struct IonHasher<D>
+pub fn hash_element<E, D>(elem: &E) -> IonResult<Output<D>>
```

Comment:
Note this can fail because we couldn't find the text for a symbol. Even though I haven't actually implemented symbols, I think this is the right API.